packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -1171,3 +1190,89 @@ bool _isForwardOrCompleted(Animation<double> animation) {
   return animation.status == AnimationStatus.forward
       || animation.status == AnimationStatus.completed;
 }
+
+NavigationBarThemeData _defaultsFor(BuildContext context) {
+  return Theme.of(context).useMaterial3 ? _TokenDefaultsM3(context) : _Defaults(context);
+}
+
+class _Defaults extends NavigationBarThemeData {
```

Comment:
Since this widget is still fairly new, if it is possible to remove these defaults that would make the code more straightforward (they dont actually follow any spec as they were sort of a in-between M2 and M3 thing)

Reply:
I wasn't sure how many people had been using this widget, so I didn't want to break anyone. I would be more than happy to just ditch the pre-M2 settings if this isn't an issue for anyone.